### PR TITLE
table partition: exchange partition feature description

### PR DESCRIPTION
**Pull request by ymkzpx**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -577,9 +577,13 @@ Empty set (0.00 sec)
 
 确保要交换入分区中的所有行与分区定义匹配；否则，你将无法找到这些行，并导致意外情况出现。
 
-> **警告：**
->
-> `EXCHANGE PARTITION` 是实验性功能，不建议在生产环境中使用。要启用 `EXCHANGE PARTITION`，将系统变量 `tidb_enable_exchange_partition` 设置为 `ON`。
+TiDB 中使用 `EXCHANGE PARTITION` 功能不仅需要满足 [MySQL 的 EXCHANGE PARTITION 条件](https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html) ，还需要满足交换分区和非分区表 Placement Rules 一样。
```
Comment (bb7133) [inline partitioned-table.md:580]: 请注意 TiDB 有一些专有特性，当表结构中包含这些特性时，在 TiDB 中使用 `EXCHANGE PARTITION` 功能不仅需要满足 [MySQL 的 EXCHANGE PARTITION 条件](https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html)，还要保证这些专有特性对于分区表和非分区表上定义相同，这些专有特性包括：

* Placement Rules in SQL（+连接）：Placement Policy 定义相同。
* TiFlash（+连接）：TiFlash Replica 数量相同。
* 聚簇索引：分区表和非分区表要么都是聚簇索引，要么都不是聚簇索引。

File: partitioned-table.md [hunk 2]
```diff
@@ -577,9 +577,17 @@ Empty set (0.00 sec)
...
+
+* [Placement Rules in SQL](/placement-rules-in-sql.md)：Placement Policy 定义相同。
+* [TiFlash](/tikv-overview.md)：TiFlash Replica 数量相同。
+* [聚簇索引](/clustered-indexes.md)：分区表和非分区表要么都是聚簇索引，要么都不是聚簇索引。
+
+此外 `EXCHANGE PARTITION` 和使用其他组件兼容性上存在一些限制，需要保证分区表和非分区表：
+
+- TiFlash：TiFlash Replica 定义不同时，无法执行 `EXCHANGE PARTITION` 操作。
+- TiCDC：都有主键或者唯一键时，TiCDC 同步 `EXCHANGE PARTITION` 操作；反之 TiCDC 将不会同步。
+- Lightning & BR：使用 Lightning、BR 导入、恢复过程中，不要执行q `EXCHANGE PARTITION` 操作。
```
Comment (hfxsd) [inline partitioned-table.md:590]:
```suggestion
- Lightning 和 BR：使用 Lightning 和 BR 导入、恢复过程中，不要执行 q `EXCHANGE PARTITION` 操作。
```

File: partitioned-table.md [hunk 3]
```diff
@@ -577,9 +577,17 @@ Empty set (0.00 sec)
...
-> **警告：**
->
-> `EXCHANGE PARTITION` 是实验性功能，不建议在生产环境中使用。要启用 `EXCHANGE PARTITION`，将系统变量 `tidb_enable_exchange_partition` 设置为 `ON`。
+请注意 TiDB 有一些专有特性，当表结构中包含这些特性时，在 TiDB 中使用 `EXCHANGE PARTITION` 功能不仅需要满足 [MySQL 的 EXCHANGE PARTITION 条件](https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html)，还要保证这些专有特性对于分区表和非分区表上定义相同，这些专有特性包括：
+
+* [Placement Rules in SQL](/placement-rules-in-sql.md)：Placement Policy 定义相同。
+* [TiFlash](/tikv-overview.md)：TiFlash Replica 数量相同。
+* [聚簇索引](/clustered-indexes.md)：分区表和非分区表要么都是聚簇索引，要么都不是聚簇索引。
+
+此外 `EXCHANGE PARTITION` 和使用其他组件兼容性上存在一些限制，需要保证分区表和非分区表：
```
Comment (hfxsd) [inline partitioned-table.md:586]:
```suggestion
此外， `EXCHANGE PARTITION` 和其他组件兼容性上存在一些限制，需要保证分区表和非分区表：
```

File: partitioned-table.md [hunk 4]
```diff
@@ -577,9 +577,17 @@ Empty set (0.00 sec)
...
+
+* [Placement Rules in SQL](/placement-rules-in-sql.md)：Placement Policy 定义相同。
+* [TiFlash](/tikv-overview.md)：TiFlash Replica 数量相同。
+* [聚簇索引](/clustered-indexes.md)：分区表和非分区表要么都是聚簇索引，要么都不是聚簇索引。
+
+此外， `EXCHANGE PARTITION` 和其他组件兼容性上存在一些限制，需要保证分区表和非分区表：
+
+- TiFlash：TiFlash Replica 定义不同时，无法执行 `EXCHANGE PARTITION` 操作。
+- TiCDC：都有主键或者唯一键时，TiCDC 同步 `EXCHANGE PARTITION` 操作；反之 TiCDC 将不会同步。
+- Lightning 和 BR：使用 Lightning 和 BR 导入、恢复过程中，不要执行 q `EXCHANGE PARTITION` 操作。
```
Comment (hfxsd) [inline partitioned-table.md:590]:
```suggestion
- Lightning 和 BR：使用 Lightning 和 BR 导入、恢复过程中，不要执行 `EXCHANGE PARTITION` 操作。
```

File: partitioned-table.md [hunk 5]
```diff
@@ -577,9 +577,17 @@ Empty set (0.00 sec)
...
-> `EXCHANGE PARTITION` 是实验性功能，不建议在生产环境中使用。要启用 `EXCHANGE PARTITION`，将系统变量 `tidb_enable_exchange_partition` 设置为 `ON`。
+请注意 TiDB 有一些专有特性，当表结构中包含这些特性时，在 TiDB 中使用 `EXCHANGE PARTITION` 功能不仅需要满足 [MySQL 的 EXCHANGE PARTITION 条件](https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html)，还要保证这些专有特性对于分区表和非分区表上定义相同，这些专有特性包括：
+
+* [Placement Rules in SQL](/placement-rules-in-sql.md)：Placement Policy 定义相同。
+* [TiFlash](/tikv-overview.md)：TiFlash Replica 数量相同。
+* [聚簇索引](/clustered-indexes.md)：分区表和非分区表要么都是聚簇索引，要么都不是聚簇索引。
+
+此外， `EXCHANGE PARTITION` 和其他组件兼容性上存在一些限制，需要保证分区表和非分区表：
+
+- TiFlash：TiFlash Replica 定义不同时，无法执行 `EXCHANGE PARTITION` 操作。
```
Comment (hfxsd) [inline partitioned-table.md:588]:
```suggestion
此外， `EXCHANGE PARTITION` 和其他组件兼容性上存在一些限制，需要保证分区表和非分区表：

- TiFlash：定义的分区表和非分区表的 TiFlash Replica 必须保持一致，如果不同则无法执行 `EXCHANGE PARTITION` 操作。
```

File: partitioned-table.md [hunk 6]
```diff
@@ -577,9 +577,17 @@ Empty set (0.00 sec)
...
+- TiCDC：都有主键或者唯一键时，TiCDC 同步 `EXCHANGE PARTITION` 操作；反之 TiCDC 将不会同步。
```
Comment (hfxsd) [inline partitioned-table.md:589]:
```suggestion
- TiCDC：分区表和非分区表都有主键或者唯一键时，TiCDC 同步 `EXCHANGE PARTITION` 操作；反之 TiCDC 将不会同步。
```

File: partitioned-table.md [hunk 7]
```diff
@@ -577,9 +577,17 @@ Empty set (0.00 sec)
...
-> **警告：**
->
-> `EXCHANGE PARTITION` 是实验性功能，不建议在生产环境中使用。要启用 `EXCHANGE PARTITION`，将系统变量 `tidb_enable_exchange_partition` 设置为 `ON`。
+请注意 TiDB 有一些专有特性，当表结构中包含这些特性时，在 TiDB 中使用 `EXCHANGE PARTITION` 功能不仅需要满足 [MySQL 的 EXCHANGE PARTITION 条件](https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html)，还要保证这些专有特性对于分区表和非分区表上定义相同，这些专有特性包括：
+
+* [Placement Rules in SQL](/placement-rules-in-sql.md)：Placement Policy 定义相同。
+* [TiFlash](/tikv-overview.md)：TiFlash Replica 数量相同。
+* [聚簇索引](/clustered-indexes.md)：分区表和非分区表要么都是聚簇索引，要么都不是聚簇索引。
+
+此外， `EXCHANGE PARTITION` 和其他组件兼容性上存在一些限制，需要保证分区表和非分区表：
```
Comment (hfxsd) [inline partitioned-table.md:586]:
```suggestion
此外， `EXCHANGE PARTITION` 和其他组件兼容性上存在一些限制，需要保证分区表和非分区表的一致性：
```

File: partitioned-table.md [hunk 8]
```diff
@@ -577,9 +577,17 @@ Empty set (0.00 sec)
 
 确保要交换入分区中的所有行与分区定义匹配；否则，你将无法找到这些行，并导致意外情况出现。
```
Comment (moyun) [inline partitioned-table.md:578]: 意外情况是指什么？能简略描述一下吗？

Comment (moyun) [reply partitioned-table.md:578]: 主要是向用户说清楚：是不可预期的行为还是什么特定的行为导致特定后果

Comment (ymkzpx) [reply partitioned-table.md]: 已经更新

File: partitioned-table.md [hunk 9]
```diff
@@ -575,11 +575,19 @@ Empty set (0.00 sec)
...
+确保要交换入分区中的所有行与分区定义匹配；否则，交换将失败。
 
-> **警告：**
->
-> `EXCHANGE PARTITION` 是实验性功能，不建议在生产环境中使用。要启用 `EXCHANGE PARTITION`，将系统变量 `tidb_enable_exchange_partition` 设置为 `ON`。
+请注意对于以下 TiDB 专有的特性，当表结构中包含这些特性时，在 TiDB 中使用 `EXCHANGE PARTITION` 功能不仅需要满足 [MySQL 的 EXCHANGE PARTITION 条件](https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html)，还要保证这些专有特性对于分区表和非分区表上定义相同。
+
+* [Placement Rules in SQL](/placement-rules-in-sql.md)：Placement Policy 定义相同。
+* [TiFlash](/tikv-overview.md)：TiFlash Replica 数量相同。
+* [聚簇索引](/clustered-indexes.md)：分区表和非分区表要么都是聚簇索引，要么都不是聚簇索引。
```
Comment (hfxsd) [inline partitioned-table.md:584]:
```suggestion
* [聚簇索引](/clustered-indexes.md)：分区表和非分区表要么都是聚簇索引 (CLUSTERED)，要么都不是聚簇索引 (NONCLUSTERED)。
```

**Review by moyun — APPROVED** [review]
LGTM

File: partitioned-table.md [hunk 10]
```diff
@@ -575,11 +575,19 @@ Empty set (0.00 sec)
 
 例如，`ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table` 交换的是 `p1` 分区的 `non_partitioned_table` 表和 `partitioned_table` 表。
```
Comment (hfxsd) [inline partitioned-table.md:576]:
```suggestion
例如，`ALTER TABLE partitioned_table EXCHANGE PARTITION p1 WITH TABLE non_partitioned_table` 交换的是 `p1` 分区的 `partitioned_table` 表和 `non_partitioned_table` 表。
```

**Review by qiancai — APPROVED** [review]
LGTM

**Review by hfxsd — APPROVED** [review]
LGTM

**Pull request merged**